mne/minimum_norm/DeFleCT.py

```diff
@@ -0,0 +1,463 @@
+# Authors: Olaf Hauk <olaf.hauk@mrc-cbu.cam.ac.uk>
+#
+# License: BSD (3-clause)
+
+from copy import deepcopy
+import numpy as np
+import scipy
+from mne.utils import logger, verbose
+from mne import pick_types, pick_channels_forward, pick_channels_cov, SourceEstimate
+from mne.cov import prepare_noise_cov 
+from mne.cov import regularize as cov_regularize
+from mne.proj import make_eeg_average_ref_proj 
+from mne.minimum_norm.inverse import _prepare_forward
+
+
+def DeFleCT_matrix(F, P, i, t, lambda2_S=1/9.):
+    """ Computes the linear DeFleCT estimator given its component matrices
+
+    Parameters:
+    -----------
+    F: leadfield/forward solution matrix (n_chan x n_vert)
+    P: projection matrix (n_chan x n_comp)
+    i: desired sensitivity to columns in P
+    t: desired CTF associated with w
+    lambda2_S: regularization parameter for Gram matrix (S)
+    (notation taken from Hauk&Stenroos HBM 2014 paper)
+
+    Returns:
+    --------
+    w: np-array, Spatial filter/linear estimator for first column of P
+
+    OH July 2015
+    """
+
+    # variable names reflect the chain of matrix operations
+    # small t: transpose; small i: (pseudo)inverse
+    # brackets and +-* operations are not reflected in variable names
+
+    # w = (tF^t + (i - tF^t S-1 P) (P^t S-1 P)-1 P^t) S-1  
+    # S = F F^t + lambda2 C
+
+    # for whitened data, covariance matrix is identity
+
+    [n_chan, n_comp] = P.shape
+    [n_chan2, n_vert] = F.shape
+    if n_chan != n_chan2:
+        raise ValueError("\nNumber of channels doesn't match between P and F: \
```

Comment:
Removed. Although I find screen output easier to follow if there are more empty lines.